packages/flame/lib/src/components/mixins/component_viewport_margin.dart

```diff
@@ -19,7 +20,8 @@ import 'package:meta/meta.dart';
 /// Do note that this only works with the old style camera and not the
 /// [CameraComponent].
 // TODO(Lukas): Rename this since it isn't necessarily related to the viewport.
-mixin ComponentViewportMargin on PositionComponent, HasGameReference {
+mixin ComponentViewportMargin<T extends FlameGame<World>>
```

Comment:
Can `World` also be a generic?

Reply:
`HasGameReference` is only `<T extends FlameGame>`. You would need another generic parameter to specify the `X extends World` requirement.  That's not needed since the type `T` is already "MyGame" which brings along all its types with it; e.g. 

 ```dart
 MyGame extends LeapGame{}
 LeapGame extends FlameGame<LeapWorld>
 ```

MyGame carries with it `FlameGame<LeapWorld>`.

What I don't need here is `<World>`